lib/src/components/DatePicker/index.tsx

```diff
@@ -62,18 +62,18 @@ export const DatePicker = forwardRef<'input', DatePickerProps>(
     ref
   ) => {
     const timeIntervals = rest?.timeIntervals
-    const formatDate = (date: string | number | Date) => getDate(date, timeIntervals)
+    const formatDate = (date: Date | number | string) => getDate(date, timeIntervals)
 
     const placeholderText = placeholder || rest.placeholderText
 
     const [focused, setFocused] = useState<Focused>((autoFocus && 'date') || null)
-    const [date, setDate] = useState(formatDate(value))
-    const inputRef = useRef<HTMLInputElement>()
+    const [date, setDate] = useState<Date | null>(formatDate(value))
+    const inputRef = useRef<HTMLInputElement | null>(null)
 
     // format date at component mount
     useEffect(() => {
-      onChange && onChange(formatDate(value))
-      //eslint-disable-next-line
+      onChange?.(formatDate(value))
```

Comment:
I think it's more readable to have a if condition here (same for all the functions in this file)